OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -712,7 +712,7 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 						targetClient.Color = SanitizePlayerColor(server, newColor, targetClient.Index, conn);
 
 						// Only update player's preferred color if new color is valid
-						if (newColor == targetClient.Color)
+						if (newColor.RGB == targetClient.Color.RGB)
```

Comment:
Please fix this inside HSLColor's `operator ==` so that everyone can benefit from the fix.